app/navigation/Navigation.scala

```diff
@@ -151,4 +156,24 @@ class Navigation @Inject()(dateTimeHelper: DateTimeHelper,
       controllers.routes.CheckYourAnswersController.onPageLoad()
     }
   }
+
+  def routeToAdditionalUploadOrCYAPage(mode: Mode, answer: Option[String]): Call = {
+    answer match {
+      case Some(ans) if ans.equalsIgnoreCase("yes") => routes.OtherReasonController.onPageLoadForAnotherFileUpload(mode)
+      case Some(ans) if ans.equalsIgnoreCase("no") => routes.CheckYourAnswersController.onPageLoad()
+      case _ =>
+        logger.debug("[Navigation][routingToAdditionalUploadOrCYAPage]: unable to get answer - reloading 'AnotherFileUploadPage'")
+        routes.OtherReasonController.onPageLoadForAnotherFileUpload(mode)
+    }
+  }
+
+  def routeForUploadList(answer: Option[String], request: UserRequest[_], mode: Mode): Call = {
+    answer match {
+      case Some(ans) if ans.equalsIgnoreCase("yes") => routes.OtherReasonController.onPageLoadForAnotherFileUpload(mode)
+      case Some(ans) if ans.equalsIgnoreCase("no") => routeToMakingALateAppealOrCYAPage(request, mode)
+      case _ =>
+        logger.debug("[Navigation][routeForUploadList]: unable to get answer - reloading 'UploadListPage'")
+        routes.OtherReasonController.onPageLoadUploadList(mode)
+    }
+  }
```

Comment:
I don't think you need `routeToAdditionalUploadOrCYAPage` as its practically duplicating `routeForUploadList`